kie-server-parent/kie-server-services/kie-server-services-openshift/src/test/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateCloudRepositoryTest.java

```diff
@@ -0,0 +1,362 @@
+        KieServerState kss = repo.load(TEST_KIE_SERVER_ID);
+
+        // Remove the configmap created at Setup to set a no pre-seeded scenario
+        client.configMaps().inNamespace(testNamespace).withName(TEST_KIE_SERVER_ID).delete();
+
+        repo.store(TEST_KIE_SERVER_ID, kss);
+    }
+
+    @Test
+    public void testAnnotation() {
+
+        String kieServer1 = KIE_SERVER_STARTUP_IN_PROGRESS_KEY_PREFIX + UUID.randomUUID().toString();
+        String kieServer2 = KIE_SERVER_STARTUP_IN_PROGRESS_KEY_PREFIX + UUID.randomUUID().toString();
+
+        ConfigMap cm = client.configMaps().withName(TEST_KIE_SERVER_ID).get();
+        ObjectMeta md = cm.getMetadata();
+        Map<String, String> ann = md.getAnnotations() == null ? new HashMap<>() : md.getAnnotations();
+        md.setAnnotations(ann);
+        ann.put(kieServer1, KIE_SERVER_STARTUP_IN_PROGRESS_VALUE);
+        ann.put(kieServer2, KIE_SERVER_STARTUP_IN_PROGRESS_VALUE);
+        client.configMaps().createOrReplace(cm);
+
+        assertNotNull(client.configMaps().withName(TEST_KIE_SERVER_ID).get().getMetadata().getAnnotations());
+        assertTrue(client.configMaps().withName(TEST_KIE_SERVER_ID).get().getMetadata()
+                         .getAnnotations().containsKey(kieServer1));
+        assertTrue(client.configMaps().withName(TEST_KIE_SERVER_ID).get().getMetadata()
+                         .getAnnotations().containsKey(kieServer2));
+        assertTrue(
+                   client.configMaps().withName(TEST_KIE_SERVER_ID).get().getMetadata()
+                         .getAnnotations().containsValue(KIE_SERVER_STARTUP_IN_PROGRESS_VALUE));
+
+        ann.remove(kieServer1);
+        client.configMaps().createOrReplace(cm);
+
+        assertTrue(
+                   client.configMaps().withName(TEST_KIE_SERVER_ID).get().getMetadata()
+                         .getAnnotations().containsValue(KIE_SERVER_STARTUP_IN_PROGRESS_VALUE));
+
+        ann.remove(kieServer2);
+        client.configMaps().createOrReplace(cm);
+
+        assertFalse(
+                    client.configMaps().withName(TEST_KIE_SERVER_ID).get().getMetadata()
+                          .getAnnotations().containsValue(KIE_SERVER_STARTUP_IN_PROGRESS_VALUE));
+
+    }
+    
+    @Test
+    public void testNPEWhenNoServiceProviderConfig() {
+        ServiceLoader<KieServerStateRepository> serverStateRepos = ServiceLoader.load(KieServerStateRepository.class);
+        assertNotNull(serverStateRepos);
+        
+        String repoType = StartupStrategyProvider.get().getStrategy().getRepositoryType();
+        for (KieServerStateRepository repo : serverStateRepos) {
+            assertNotNull(repo);
+            assertNotNull(repo.getClass().getSimpleName());
+            if (repo.getClass().getSimpleName().equals(repoType)) {
+                fail("Unexpected repo type: " + repoType);
+            }
+        }
+    }
+    
+    @Test
+    public void testCreateAndLoad() throws InterruptedException {
```

Comment:
What is the purpose of this test?

Reply:
To test the new 'create' method added to KieServerStateCloudRepository. This new method is supporting KieServerTemplateStorage OpenShift implementation.